realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -203,7 +202,12 @@ private void postError(final ObjectServerError error) {
                     handler.post(new Runnable() {
                         @Override
                         public void run() {
-                            callback.onError(error);
+                            try {
+                                callback.onError(error);
+                            } catch (Exception e) {
+                                RealmLog.info("onError has thrown an exception but ignoring it: %s", e.getMessage());
```

Comment:
"but is ignoring it."

Also use `io.realm.internal.Util.getStackTrace(Throwable)` instead of `e.getMessage()`. That will show the entire stack trace in logcat.